### ISSUE TITLE
Black screen after boot

### ISSUE THREAD

**Issue by diversys** [SerenityOS/serenity#6670]
This is ba9a285d56ec364e3e5b63e2f3c1e7677bfaa738 but started a few days ago.
Host is macOS 10.14.6

```
0/2] Re-checking globbed directories...
[1/2] Re-running CMake...
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/diver/serenity/Build
[0/2] Re-checking globbed directories...
[22/2278] Linking CXX static library Meta/Lagom/libLagomCore.a
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: Meta/Lagom/libLagomCore.a(Command.cpp.o) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: Meta/Lagom/libLagomCore.a(FileWatcher.cpp.o) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: Meta/Lagom/libLagomCore.a(Command.cpp.o) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: Meta/Lagom/libLagomCore.a(FileWatcher.cpp.o) has no symbols
[1715/1716] Install the project...
-- Install configuration: ""
[0/2] Re-checking globbed directories...
[0/1] cd /Users/diver/serenity/Build && /usr/local/Cellar/cmake/3.20.1/...enity SERENITY_ARCH=i686 /Users/diver/serenity/Meta/build-image-qemu.sh
checking existing image
e2fsck 1.46.2 (28-Feb-2021)
Pass 1: Checking inodes, blocks, and sizes
Pass 2: Checking directory structure
Pass 3: Checking directory connectivity
Pass 3A: Optimizing directories
Pass 4: Checking reference counts
Pass 5: Checking group summary information

_disk_image: ***** FILE SYSTEM WAS MODIFIED *****
_disk_image: 5631/46736 files (1.0% non-contiguous), 84582/186368 blocks
done
mounting filesystem... done
installing base system... done
creating initial filesystem structure... done
creating utmp file... done
setting up device nodes folder... done
writing version file... done
installing users... done
adding some desktop icons...done
installing shortcuts... done
installing 'checksum' variants... done
unmounting filesystem... done
[0/2] Re-checking globbed directories...
[0/1] cd /Users/diver/serenity/Build && /Users/diver/serenity/Meta/run.sh
qemu-system-x86_64: multiboot knows VBE. we don't
[Kernel]: CPU[0]: Supported features: nx pae pge rdrand smep sse tsc constant_tsc sep syscall mmx sse2 sse3 ssse3 sse4.1 sse4.2 xsave avx
[Kernel]: CPU[0]: Physical address bit width: 36
[#0 Kernel]: Installing Unhandled Handlers
[Kernel]: MM: boot_pdpt @ P0x0010a000
[Kernel]: MM: boot_pd0 @ P0x0010b000
[Kernel]: MM: boot_pd3 @ P0x0010c000
[Kernel]: MM: Low memory range @ P0x00000000 - P0x00100000
[Kernel]: MM: Kernel range @ P0x00100000 - P0x00f3c000
[Kernel]: MM: Multiboot mmap: address=0x00000000, length=654336, type=1
[Kernel]: MM: Got an unaligned physical_region from the bootloader; correcting length 654336 by 3072 bytes
[Kernel]: MM: Multiboot mmap: address=0x0009fc00, length=1024, type=2
[Kernel]: MM: Multiboot mmap: address=0x000f0000, length=65536, type=2
[Kernel]: MM: Multiboot mmap: address=0x00100000, length=535691264, type=1
[Kernel]: MM: Multiboot mmap: address=0x1ffe0000, length=131072, type=2
[Kernel]: MM: Multiboot mmap: address=0xfffc0000, length=262144, type=2
[Kernel]: MM: Super physical region: P0x00e3c000 - P0x00f3c000
[Kernel]: MM: User physical region: P0x00f3d000 - P0x1ffe0000
[Kernel]: MM: Contiguous reserved range from P0x0009fc00, length is 394240
[Kernel]: MM: Contiguous reserved range from P0x1ffe0000, length is 3758227456
[#0 Kernel]: Early access to ACPI tables for interrupt setup
[Kernel]: Interrupts: Switch to Legacy PIC mode
[Kernel]: PIC: Cascading mode, vectors 0x50-0x5f
[#0 Kernel]: Trying to unregister unused handler (?)
[#0 Kernel]: Trying to unregister unused handler (?)
[#0 Kernel]: Interrupts: Detected Dual i8259
[Kernel]: ACPI: Using RSDP @ P0x000f5850
[Kernel]: ACPI: Main Description Table valid? true
[Kernel]: ACPI: Using RSDT, enumerating tables @ P0x1ffe15ab
[Kernel]: ACPI: RSDT revision 1, total length: 52
[Kernel]: ACPI: Initializing Fixed ACPI data
[Kernel]: ACPI: Searching for the Fixed ACPI Data Table
[Kernel]: ACPI: Fixed ACPI data, Revision 1, length: 116 bytes
[Kernel]: ACPI: DSDT P0x1ffe0040
[Kernel]: ACPI: Dynamic Parsing Enabled, Can parse AML
[Kernel]: Starting SerenityOS...
[Kernel]: RTC: Year: 2021, month: 4, day: 26, hour: 16, minute: 39, second: 41
[Kernel]: HPET @ P0x1ffe154b
[Kernel]: HPET: Minimum clock tick - 0
[Kernel]: HPET: Timers count - 3
[Kernel]: HPET: Main counter size: 64-bit
[Kernel]: HPET: Timer[0] comparator size: 64-bit, mode: 64-bit
[Kernel]: HPET: Timer[1] comparator size: 64-bit, mode: 64-bit
[Kernel]: HPET: Timer[2] comparator size: 64-bit, mode: 64-bit
[Kernel]: HPET: frequency 100000000 Hz (100 MHz) resolution: 10 ns
[#0 Kernel]: Trying to unregister unused handler (?)
[#0 Kernel]: Trying to unregister unused handler (?)
[#0 Kernel]: HPET: Setting appropriate functions to timers.
[#0 Kernel]: Time: Scanning for periodic timers
[#0 Kernel]: Time: Scanning for non-periodic timers
[Kernel]: KernelRng: Using RDSEED or RDRAND as entropy source
[Kernel]: VMWareBackdoor: Enabling absolute mouse mode
[#0 Kernel]: I8042: Dual channel controller
[Kernel]: PS2MouseDevice: Mouse wheel enabled!
[Kernel]: PS2MouseDevice: 5 buttons enabled!
[#0 Kernel]: Trying to unregister unused handler (?)
[#0 Kernel]: Trying to unregister unused handler (?)
[#0 Kernel]: VC: Switch to 0 (0xc0d827b4)
[#0 colonel(0:0)]: Scheduler[0]: idle loop running
[init_stage2(2:2)]: PCI: Using I/O instructions for PCI configuration space access
[init_stage2(2:2)]: PCI [0000:00:00:00] PCI::ID [8086:1237]
[init_stage2(2:2)]: PCI [0000:00:01:00] PCI::ID [8086:7000]
[init_stage2(2:2)]: PCI [0000:00:01:01] PCI::ID [8086:7010]
[init_stage2(2:2)]: PCI [0000:00:01:02] PCI::ID [8086:7020]
[init_stage2(2:2)]: PCI [0000:00:01:03] PCI::ID [8086:7113]
[#0 SyncTask(3:3)]: SyncTask is running
[init_stage2(2:2)]: PCI [0000:00:02:00] PCI::ID [1234:1111]
[init_stage2(2:2)]: PCI [0000:00:03:00] PCI::ID [8086:2922]
[init_stage2(2:2)]: PCI [0000:00:04:00] PCI::ID [1af4:1003]
[init_stage2(2:2)]: PCI [0000:00:05:00] PCI::ID [1af4:1005]
[init_stage2(2:2)]: PCI [0000:00:06:00] PCI::ID [8086:100e]
[#0 init_stage2(2:2)]: BXVGA: framebuffer @ P0xf8000000
[#0 init_stage2(2:2)]: BXVGADevice resolution set to 1024x768 (pitch=4096)
[init_stage2(2:2)]: UHCI: Controller found PCI::ID [8086:7020] @ PCI [0000:00:01:02]
[init_stage2(2:2)]: UHCI: I/O base IO c080
[init_stage2(2:2)]: UHCI: Interrupt line: 11
[#0 init_stage2(2:2)]: UHCI: Allocated framelist at physical address P0x00e3c000
[#0 init_stage2(2:2)]: UHCI: Framelist is at virtual address V0xc115d000
[#0 init_stage2(2:2)]: UHCI: QH(0xc115f000) @ 14929920: link_ptr=14929954, element_link_ptr=1
[#0 init_stage2(2:2)]: UHCI: QH(0xc115f020) @ 14929952: link_ptr=14929986, element_link_ptr=1
[#0 init_stage2(2:2)]: UHCI: QH(0xc115f040) @ 14929984: link_ptr=14930018, element_link_ptr=1
[#0 init_stage2(2:2)]: UHCI: QH(0xc115f060) @ 14930016: link_ptr=14930050, element_link_ptr=1
[#0 init_stage2(2:2)]: UHCI: QH(0xc115f080) @ 14930048: link_ptr=14942209, element_link_ptr=1
[#0 init_stage2(2:2)]: UHCI: Reset completed
[#0 init_stage2(2:2)]: UHCI: Started
[#0 init_stage2(2:2)]: DMIExpose: SMBIOS 32bit Entry point @ P0x000f5870
[#0 init_stage2(2:2)]: DMIExpose: Data table @ P0x000f5890
[#0 init_stage2(2:2)]: VirtIOConsole: Found @ PCI [0000:00:04:00]
[#0 init_stage2(2:2)]: Trying to unregister unused handler (?)
[#0 init_stage2(2:2)]: VirtIOConsole: Multi port is not yet supported!
[#0 init_stage2(2:2)]: VirtIOConsole: cols: 0, rows: 0, max nr ports 0
[#0 init_stage2(2:2)]: VirtIORNG: Found @ PCI [0000:00:05:00]
[#0 init_stage2(2:2)]: Trying to unregister unused handler (?)
[init_stage2(2:2)]: E1000: Found @ PCI [0000:00:06:00]
[init_stage2(2:2)]: E1000: port base: IO c040
[init_stage2(2:2)]: E1000: MMIO base: P0xfebc0000
[init_stage2(2:2)]: E1000: MMIO base size: 131072 bytes
[init_stage2(2:2)]: E1000: Interrupt line: 10
[init_stage2(2:2)]: E1000: Has EEPROM? true
[init_stage2(2:2)]: E1000: MAC address: 52:54:00:12:34:56
[#0 init_stage2(2:2)]: IDE controller @ PCI [0000:00:01:01]: bus master base was set to IO c0e0
[#0 init_stage2(2:2)]: IDE controller @ PCI [0000:00:01:01]: interrupt line was set to 0
[#0 init_stage2(2:2)]: IDE controller @ PCI [0000:00:01:01]: ISA Compatibility mode-only controller, supports bus mastering
[#0 init_stage2(2:2)]: IDE controller @ PCI [0000:00:01:01]: primary channel DMA capable? 0
[#0 init_stage2(2:2)]: IDE controller @ PCI [0000:00:01:01]: secondary channel DMA capable? 0
[#0 init_stage2(2:2)]: IDEChannel: Primary master ATA device found: Name=QEMU HARDDISK, Capacity=190840832, Capabilities=0x0b00
[#0 init_stage2(2:2)]: Trying to unregister unused handler (?)
[#0 init_stage2(2:2)]: IDEChannel: Secondary master device appears to be ATAPI. We're going to ignore it for now as we don't support it.
[init_stage2(2:2)]: PCI [0000:00:03:00]: AHCI controller reset
[#0 init_stage2(2:2)]: PCI [0000:00:03:00]: AHCI command list entries count - 32
[init_stage2(2:2)]: AHCI Port 1: Device not detected, Phy not enabled
[init_stage2(2:2)]: AHCI Port 2: Device not detected, Phy not enabled
[init_stage2(2:2)]: AHCI Port 3: Device not detected, Phy not enabled
[init_stage2(2:2)]: AHCI Port 4: Device not detected, Phy not enabled
[init_stage2(2:2)]: AHCI Port 5: Device not detected, Phy not enabled
[init_stage2(2:2)]: AHCI Port 6: Device not detected, Phy not enabled
[#0 init_stage2(2:2)]: Master Boot Record: invalid signature
[init_stage2(2:2)]: VFS: mounted root from Ext2FS (device:3,0 (PATADiskDevice))
[init_stage2(2:2)]: Loading kernel symbol table...
[init_stage2(2:2)]: Unmapped 352 KiB of kernel text after init! :^)
[NetworkTask(7:7)]: NetworkTask: LoopbackAdapter network adapter found: hw=13:55:02:09:55:aa
[NetworkTask(7:7)]: NetworkTask: E1000NetworkAdapter network adapter found: hw=52:54:00:12:34:56

```

**Comment by linusg** [comment]
cc @nooga

**Comment by nooga** [comment]
Same situation :^(

**Comment by jagger27** [comment]
Seeing this on Big Sur too. Working fine on my Linux machine though.

**Comment by willmcpherson2** [comment]
Also having this issue on aarch64 Big Sur. What's the significance of https://github.com/SerenityOS/serenity/commit/ba9a285d56ec364e3e5b63e2f3c1e7677bfaa738? Did you get their via git bisect?

**Comment by diversys** [comment]
Fixed in 8ae07945845eab0cb350d6f72bd6eb5a12766114.
